source/mqtt/MqttClient.cpp

```diff
@@ -295,6 +295,41 @@ namespace Aws
                 }
             }
 
+            void MqttConnection::s_connectionInit(
+                MqttConnection *self,
+                const char *hostName,
+                uint16_t port,
+                const Io::SocketOptions &socketOptions,
+                aws_mqtt5_client *mqtt5client)
+            {
+
+                self->m_hostName = String(hostName);
+                self->m_port = port;
+                self->m_socketOptions = socketOptions;
+
+                self->m_underlyingConnection = aws_mqtt_client_connection_new_from_mqtt5_client(mqtt5client);
+
+                if (self->m_underlyingConnection)
```

Comment:
This section seems identical to the one for the function being overloaded. Is it worth it to split this off into a small helper function that both `s_connectionInit` call? I don't think we even need to pass the function an argument since they'd be using `self->m_underlyingConnection`. Also, there is a way for `self->m_underlyingConnection` to be null so should we at least log an error or something in that case? Currently both do nothing at this point if it's null.